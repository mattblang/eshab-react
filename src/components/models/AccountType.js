// @flow

class AccountType {
    static BUDGET: number = 0
    static OFF_BUDGET: number = 1
    static CLOSED: number = 2
}

export default AccountType