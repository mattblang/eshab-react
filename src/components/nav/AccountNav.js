// @flow

import React, {Component} from 'react'
import {Link} from 'react-router'
import * as firebase from 'firebase'
import Account from '../models/Account'
import AccountType from '../models/AccountType'

type AccountNavProps = {
    type: number
}
type AccountNavState = {
    accounts: Array<Account>
}

class AccountNav extends Component {

    props: AccountNavProps
    state: AccountNavState
    accountsRef = {}
    accountInput = {}

    constructor(props: AccountNavProps) {
        super(props)
        this.state = {
            accounts: []
        }
    }

    componentDidMount() {
        // this.accountsRef = firebase.database().ref('accounts')
        // this.accountsRef.orderByChild('type').equalTo(this.props.type).on('value', snapshot => {
        //     const val = snapshot.val() || {}
        //     this.setState({
        //         accounts: Object.keys(val).map((key) => {
        //             return Account.parseFirebase(val[key], key)
        //         })
        //     })
        // })
    }

    render() {
        return (
            <nav style={styles.container}>
                <div><strong>{this.getLabel()}</strong></div>
                {this.state.accounts.map((account, i) =>
                    <div key={i}>
                        <Link to={`/account/${account.id}`}>{account.name}</Link>
                        <button onClick={() => this.removeAccount(account)}>Delete</button>
                    </div>
                )}
                <input
                    ref={(account) => {
                        this.accountInput = account
                    }}
                    type="text"
                    placeholder="Enter account"/>
                <input onClick={() => this.addAccount()} type="button" value="Add Account"/>
            </nav>
        )
    }

    getLabel() {
        switch (this.props.type) {
            case AccountType.BUDGET:
                return 'Budget Accounts'
            case AccountType.OFF_BUDGET:
                return 'Off-Budget Accounts'
            case AccountType.CLOSED:
                return 'Closed Accounts'
            default:
                return null
        }

    }

    addAccount() {
        this.accountsRef.push({
            name: this.accountInput.value,
            type: this.props.type
        })
        this.accountInput.value = null
    }

    removeAccount(account: Account) {
        this.accountsRef.child(account.id).remove()
    }
}

let styles = {
    container: {
        backgroundColor: 'blue',
        color: 'red'
    }
}

export default AccountNav