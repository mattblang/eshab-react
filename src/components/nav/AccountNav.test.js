import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import AccountNav from './AccountNav'
import AccountType from '../models/AccountType'

beforeAll(() => {
    AccountNav.prototype.setupFirebase = jest.fn()
})

it('renders', () => {
    const component = shallow(
        <AccountNav type={AccountType.BUDGET} />
    )
    expect(toJson(component)).toMatchSnapshot()
})

it('displays appropriate title', () => {
    const component = shallow(
        <AccountNav type={AccountType.BUDGET} />
    )
    expect(component.find('strong').text()).toEqual('Budget Accounts')
})
