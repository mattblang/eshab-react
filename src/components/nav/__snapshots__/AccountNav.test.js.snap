exports[`test renders correctly 1`] = `
<nav
  style={
    Object {
      "backgroundColor": "blue",
      "color": "red",
    }
  }>
  <div>
    <strong>
      Budget Accounts
    </strong>
  </div>
  <input
    placeholder="Enter account"
    type="text" />
  <input
    onClick={[Function]}
    type="button"
    value="Add Account" />
</nav>
`;
